include/ginkgo/core/solver/solver_base.hpp

```diff
@@ -48,6 +50,27 @@ namespace gko {
 namespace solver {
 
 
+/**
+ * Traits class providing information on the type and location of workspace
+ * vectors inside a solver.
+ */
+template <typename Solver>
+struct solver_workspace_traits {
+    // number of vectors used by this workspace
+    static int num_vectors(const Solver&) { return 0; }
+    // number of arrays used by this workspace
+    static int num_arrays(const Solver&) { return 0; }
+    // array containing the num_vectors names for the workspace vectors
```

Comment:
here 'vectors' means both vectors and scalars, while below these are separated. Perhaps `linop_names` is better?

Reply:
op_names?